pkg/prometheus/promcfg_test.go

```diff
@@ -634,6 +635,369 @@ alerting:
 	}
 }
 
+func TestSettingHonorTimestampsInServiceMonitor(t *testing.T) {
+	cg := &configGenerator{}
+	cfg, err := cg.generateConfig(
+		&monitoringv1.Prometheus{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "default",
+			},
+			Spec: monitoringv1.PrometheusSpec{
+				Version: "v2.9.0",
```

Comment:
I don't remember this needing to be here when I reviewed last week? 🤔

Reply:
This is now necessary due to https://github.com/coreos/prometheus-operator/pull/2800#discussion_r336875968